src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -11,6 +11,28 @@
 
 namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
+    internal enum NullableAnnotation : byte
+    {
+        Unknown,     // No information. Think oblivious.
+        NotNullable, // For string, int, T
+        Nullable,    // For string?, T? where T : class; and for int?, T? where T : struct.
+        NotNullableBasedOnAnalysis, // Explicitly set by flow analysis
+        NullableBasedOnAnalysis, // Explicitly set by flow analysis
+    }
+
+    internal static class NullableAnnotationExtensions
+    {
+        public static bool IsAnyNullable(this NullableAnnotation annotation)
```

Comment:
>IsAny [](start = 27, length = 5)

I trip over `IsAny` when I see these methods called since `Any` is usually used for a collection. Consider naming these methods `IsNullable()` and `IsNotNullable()`.

Reply:
> I trip over IsAny when I see these methods called since Any is usually used for a collection. Consider naming these methods IsNullable() and IsNotNullable().

This is exactly the effect I want, I want people to think and understand what question these methods are answering. I think that the proposed names are going to be misleading, they would almost match specific enum values.

---
In reply to: [232112037](https://github.com/dotnet/roslyn/pull/30913#discussion_r232112037) [](ancestors = 232112037)